File: 2496-maximum-value-of-a-string-in-an-array/2496-maximum-value-of-a-string-in-an-array.cpp
class Solution {
public:
    int maximumValue(vector<string>& strs) {
        int ans=0;
        for(string s:strs) {
            bool digit=true;
            for(char c:s) {
                if(c<'0' || c>'9') {
                    digit=false;
                    break;
                }
            }
            int value;
            if(digit){
                value=stoi(s);
            }
            else {
                value=s.length();
            }
            ans=max(ans, value);
        }
        return ans;
    }
};

File: 2496-maximum-value-of-a-string-in-an-array/2496-maximum-value-of-a-string-in-an-array.py
class Solution:
    def maximumValue(self, strs: list[str]) -> int:
        ans = 0

        for s in strs:
            if s.isdigit():
                value = int(s)
            else:
                value = len(s)

            ans = max(ans, value)

        return ans